src/main/java/org/openapitools/jackson/nullable/JsonNullable.java

```diff
@@ -84,6 +157,141 @@ public void ifPresent(
         }
     }
 
+    /**
+     * If a value is present, performs the given action with the value,
+     * otherwise performs the given empty-based action.
+     *
+     * @param action the action to be performed, if a value is present
+     * @param undefinedAction the empty-based action to be performed, if no 
+     *        value is present
+     * @throws NullPointerException if a value is present and the given action
+     *         is null, or no value is present and the given empty-based action
+     *         is null
+     *
+     * @since 0.2.8
+     */
+    public void ifPresentOrElse( Consumer<? super T> action, Runnable undefinedAction ) {
+        if (this.isPresent) {
+            action.accept(value);
+        }
+        else {
+            undefinedAction.run();
+        }
+    }
+
+    /**
+     * If a value is present, and the value matches the given predicate, returns
+     *  a JsonNullable describing the value, otherwise returns an undefined
+     * JsonNullable.
+     *
+     * @param predicate the predicate to apply to a value, if present
+     * @return a JsonNullable describing the value of this JsonNullable,
+     *   if a value is present and the value matches the given predicate,
+     *   otherwise an undefined JsonNullable
+     * @throws NullPointerException if the predicate is null
+     *
+     * @since 0.2.8
+     */
+    public JsonNullable<T> filter( Predicate<T> predicate ) {
+      if (this.isPresent && predicate.test(value)) {
+        return this;
+      }
+      else {
+        return undefined();
+      }
+    }
+
+    /**
+     * If a value is present, returns a JsonNullable describing the result of
+     * applying the given mapping function to the value, otherwise returns an
+     * undeined JsonNullable.
+     *
+     * @param <U> the type of the value returned from the mapping function
+     * @param mapper the mapping function to apply to a value, if present
+     * @return a JsonNullable describing the result of applying a mapping
+     *         function to the value of this JsonNullable, if a value is
+     *         present, otherwise an undefined JsonNullable
+     * @throws NullPointerException if the mapping function is null
```

Comment:
Similar here. This only happens if the value is present. Either the description should be updated or the method.

Reply:
Probably better to update the method and just be consistent with the Optional behavior.

Reply:
Test and fix for this issue is complete.